src/main/java/com/PBA/budgetservice/service/CurrencyService.java

```diff
@@ -0,0 +1,8 @@
+package com.PBA.budgetservice.service;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+public interface CurrencyService {
```

Comment:
```suggestion
public interface CurrencyGateway {
```
Name it as a `gateway` since it's an external call.